src/compas/datastructures/mesh/duality.py

```diff
@@ -24,24 +23,41 @@ def mesh_dual(mesh, cls=None, boundary=0):
         Defaults to the type of the provided mesh object.
     boundary: float, optional
         boundary mode for the dual mesh
-        Default mode is 0, not create faces on boundaries
+        Default mode is 0, not create faces on boundaries.
+        1, create faces on mesh edges, not include original mesh boundary vertices.
+        2. create faces on mesh edges and include original mesh boundary vertices on the corner.
+        3. create faces on mesh edges and include all original mesh boundary vertices.
```

Comment:
in my opinion there are only three options

1. don't include the boundary
2. include the boundary without the original vertices
3. include the boundary with the original vertices

```python
def mesh_dual(mesh, cls=None, include_boundary=False, preserve_boundary_vertices=False):
```

Reply:
@brgcode So this one doesn't make sense? It only boundary vertices on the corner, not on edges. 
<img width="1312" alt="Screenshot 2022-05-02 at 11 32 32" src="https://user-images.githubusercontent.com/1162842/166214164-e2d3757b-dd7a-482c-9082-0cbffb0f288e.png">

Reply:
i would indeed say it only makes sense because of the specific boundary conditions. but not sure. have to think about it...

Reply:
ok, I just keep this option for now as `preserve_corner_vertices` in case someone needs it. 
now it's 
```python 
def mesh_dual(mesh, cls=None, include_boundary=False, preserve_boundary_vertices=False, preserve_corner_vertices=False):
```

Reply:
@ricardoavelino see my latest comments below.

Reply:
Just saw. Yes, indeed your PR cover all cases